neurom/features/population.py

```diff
@@ -78,6 +79,9 @@ def sholl_frequency(morphs, neurite_type=NeuriteType.all, step_size=10, bins=Non
                         for n in m.neurites if neurite_filter(n))
         bins = np.arange(min_soma_edge, min_soma_edge + max_radii, step_size)
 
+    for morph in morphs:
+        _assert_soma_center(morph, "sholl_frequency")
```

Comment:
I would opt for making this lazy, i.e. specifying an internal func:
```python
def func(morph):
    _assert_soma_center(morph, "sholl_frequency")
   return sholl_crossings(...)
```

Reply:
good idea, I changed.

Reply:
I also added the morph name in the exception